### PR TITLE
feat(editors/104): Show/edit subnetworks without redundancy

### LINKED ISSUE #649: 104 support: Show/edit subnetworks without redundancy
As an engineer 
I want to be able to see 104 communication subnetworks
So I can use it to configure devices or check the configuration.

More information can be found in IEC/TS 61850-80-1

Mockup:
104mockups-Show subnetworks drawio

The layout can be re-used for the exsisting 

Just like in the communications tab, the private 104 subnetworks should be shown to the user in the 104 plugin. This can either be a section with one link, or a section with redundant links. An example can be seen here:
 
image
image

104mockups-Edit communication (no redundancy groups) drawio

When the user clicks on one AP the `<P>` elements should be shown to the user. It should also show the descriptive names as well.
 
image

### PR DESCRIPTION
**Pull request by Flurb**
Closes #649

### CODE REVIEW

File: src/editors/protocol104/network-container.ts [hunk 1]
```diff
@@ -1,13 +1,63 @@
-import {css, customElement, html, LitElement, TemplateResult} from "lit-element";
+import {css, customElement, html, LitElement, property, TemplateResult} from "lit-element";
```
Comment (dlabordus) [inline src/editors/protocol104/network-container.ts:1]: Maybe make the formatting of the imports nicer and in line with standard way.

File: src/editors/protocol104/network-container.ts [hunk 2]
```diff
@@ -1,13 +1,63 @@
...
 
 @customElement('network-104-container')
 export class Network104Container extends LitElement {
+  @property()
+  doc!: XMLDocument;
+
+  private getSubNetworkElements(): Element[] {
+    return Array.from(this.doc.querySelectorAll('Communication > SubNetwork') ?? [])
+      .filter(network => network.getAttribute('type') === '104')
+      .sort((network1, network2) => (getNameAttribute(network1) ?? '').localeCompare(getNameAttribute(network2) ?? '') )
```
Comment (dlabordus) [inline src/editors/protocol104/network-container.ts:16]: Found this function that's used in more places:
`.sort((a,b) => compareNames(a,b));`

File: src/editors/protocol104/subnetwork-container.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  html,
+  property,
+  css,
+} from 'lit-element';
+
+import '@material/mwc-icon-button';
+
+import './connectedap-editor.js';
+import {
+
```
Comment (dlabordus) [inline src/editors/protocol104/subnetwork-container.ts:14]: Empty line can be removed.

File: src/editors/protocol104/wizards/connectedap.ts [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+import '@material/mwc-list/mwc-check-list-item';
+import '@material/mwc-icon';
+import { Checkbox } from '@material/mwc-checkbox';
+import { List } from '@material/mwc-list';
+import { ListItemBase } from '@material/mwc-list/mwc-list-item-base';
+
+import '../../../wizard-textfield.js';
+import '../../../filtered-list.js';
+import { pTypes104, stationTypeOptions, typeDescriptiveNameKeys } from '../foundation/p-types.js';
+import { compareNames, ComplexAction, createElement, EditorAction, getValue, identity, isPublic, Wizard, WizardActor, WizardInputElement } from '../../../foundation.js';
```
Comment (dlabordus) [inline src/editors/protocol104/wizards/connectedap.ts:16]: Maybe make the formatting of the imports nicer and in line with standard way.

**Review by dlabordus — CHANGES_REQUESTED** [review]

File: src/editors/protocol104/connectedap-editor.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+import '@material/mwc-fab';
+
+import '../../action-icon.js';
+import { newWizardEvent, newActionEvent } from '../../foundation.js';
+import { editConnectedAp104Wizard } from './wizards/connectedap.js';
+
+/** [[`104`]] subeditor for a `ConnectedAP` element. */
+@customElement('connectedap-104-editor')
+export class ConnectedAPEditor extends LitElement {
```
[the remark this answers is not in the document]
Comment (Flurb) [reply src/editors/protocol104/connectedap-editor.ts:17]: Will do, `ied-container` for example uses the same principle.

File: src/editors/protocol104/network-container.ts (hunk 2 above)
Comment (Flurb) [reply src/editors/protocol104/network-container.ts:16]: Ah yes, I knew that one actually. Good point!

**Pull request merged**